.github/workflows/development.yaml

```diff
@@ -0,0 +1,118 @@
+name: Pylint and Pytest Workflow
+
+permissions:
+  contents: write
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pylint
+
+      - name: Run pylint and save score
+        id: pylint
+        run: |
+          pylint **/*.py --fail-under=8 > pylint-report.txt || true
+          score=$(grep -oP '(?<=Your code has been rated at )[^/]*' pylint-report.txt)
+          echo "PYLINT_SCORE=$score" >> $GITHUB_ENV
+
+      - name: Generate badge
+        run: |
+          mkdir -p .github/badges
+          score=${{ env.PYLINT_SCORE }}
+          color="green"
+          if (( $(echo "$score < 8" | bc -l) )); then
+            color="yellow"
+          fi
+          if (( $(echo "$score < 5" | bc -l) )); then
+            color="red"
+          fi
+          echo '<svg xmlns="http://www.w3.org/2000/svg" width="160" height="40" role="img" aria-label="Pylint Results"><linearGradient id="a" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><rect rx="3" width="150" height="20" fill="#555"/><rect rx="3" x="90" width="60" height="20" fill="'$color'"/><path fill="'$color'" d="M90 0h4v20h-4z"/><rect rx="3" width="150" height="20" fill="url(#a)"/><g fill="#fff" text-anchor="middle" font-family="Verdana,DejaVu Sans,sans-serif" font-size="11"><text x="45" y="15" fill="#010101" fill-opacity=".3">Pylint</text><text x="45" y="14">Pylint</text><text x="120" y="15" fill="#010101" fill-opacity=".3">'$score'/10</text><text x="120" y="14">'$score'/10</text></g></svg>' > .github/badges/pylint.svg
```

Comment:
What exactly does this do?

Reply:
It formats the badge with the next and the sizes, and then includes the score so it can be stored into the badges dir.